hivemind/utils/mpfuture.py

```diff
@@ -303,7 +303,7 @@ def __del__(self):
     def __getstate__(self):
         return dict(
             _sender_pipe=self._sender_pipe,
-            _shared_state_code=self._shared_state_code,
+            _shared_state_code=ForkingPickler.dumps(self._shared_state_code).tobytes(),
```

Comment:
Despite the name, it does work well with all multiprocessing start methods (`fork`, `spawn`, and `forkserver`). This may be tested using this short script:

```python
import multiprocessing as mp
import torch
import time

torch.multiprocessing.set_sharing_strategy("file_system")

def foo(q):
    t = torch.zeros(3).share_memory_()
    q.put(mp.reduction.ForkingPickler.dumps(t).tobytes())
    t[1] = 2

if __name__ == '__main__':
    mp.set_start_method('spawn')
    q = mp.Queue()
    p = mp.Process(target=foo, args=(q,))
    p.start()
    time.sleep(1)
    print(mp.reduction.ForkingPickler.loads(q.get()))
    p.join()
```